## Reading, writing and displaying images

In [2]:
import cv2

In [3]:
import numpy as np

In [2]:
# Load first image
# Don't need to load library again but, it's a good habit

import cv2
import numpy as np

# Load an image using the speciafying image path
input = cv2.imread('./images/image.jpg')

# The file 'image.jpg' is loaded to input variable in python


# To display image, use 'imshow'
cv2.imshow('Hello wolrd', input)

# 'waitKey' allow show input information when image window open
# Leaving a blank key -> wait for any key press
# Insert number (milisecond) to specify a delay for how long the window opens.
cv2.waitKey(3000)

# Make sure to insert this line
# to close all windows open. Unless, the image window will cause to hang.
cv2.destroyAllWindows()

## Take a look, how images are stored

In [3]:
import numpy as np
print input.shape

(959L, 719L, 3L)


**input has 3 dimensionals**:

- Height = 959 pixels
- Width = 719 pixels
- Color = 3 colors (RGB)

## Save images to file

In [5]:
import cv2
cv2.imwrite('./images/output.jpg', input)
cv2.imwrite('./images/output.png', input)

True